packages/shopify/src/utils/metafields.ts

```diff
@@ -0,0 +1,118 @@
+import type { MetafieldType } from '../types/metafields'
+
+export const parseJson = (value: string): any => {
+  try {
+    return JSON.parse(value)
+  } catch (e) {
+    return value
+  }
+}
+
+const unitConversion: Record<string, string> = {
+  mm: 'millimeter',
+  cm: 'centimeter',
+  m: 'meter',
+  in: 'inch',
+  ft: 'foot',
+  yd: 'yard',
+
+  ml: 'milliliter',
+  l: 'liter',
+  us_fl_oz: 'fluid-ounce',
+  us_gal: 'gallon',
+
+  kg: 'kilogram',
+  g: 'gram',
+  lb: 'pound',
+  oz: 'ounce',
+
+  MILLIMETERS: 'millimeter',
+  CENTIMETERS: 'centimeter',
+  METERS: 'meter',
+
+  MILLILITERS: 'milliliter',
+  LITERS: 'liter',
+  FLUID_OUNCES: 'fluid-ounce',
+  IMPERIAL_FLUID_OUNCES: 'fluid-ounce',
+  GALLONS: 'gallon',
+
+  KILOGRAMS: 'kilogram',
+  GRAMS: 'gram',
+  OUNCES: 'ounce',
+  POUNDS: 'pound',
+
+  FEET: 'foot',
+}
+
+export const getMeasurment = (input: string, locale: string = 'en-US') => {
+  try {
+    let { unit, value } = JSON.parse(input)
+    return new Intl.NumberFormat(locale, {
+      unit: unitConversion[unit],
+      style: 'unit',
+    }).format(parseFloat(value))
+  } catch (e) {
+    console.error(e)
+    return input
+  }
+}
+
+export const getMetafieldValue = (
+  type: MetafieldType,
+  value: string,
+  locale: string = 'en-US'
+) => {
+  switch (type) {
+    case 'boolean':
+      return value === 'true' ? '&#10003;' : '&#10005;'
+    case 'number_integer':
+      return parseInt(value).toLocaleString(locale)
+    case 'number_decimal':
+      return parseFloat(value).toLocaleString(locale)
+    case 'date':
+      return Intl.DateTimeFormat(locale, {
+        dateStyle: 'medium',
+      }).format(new Date(value))
+    case 'date_time':
+      return Intl.DateTimeFormat(locale, {
+        dateStyle: 'medium',
+        timeStyle: 'long',
+      }).format(new Date(value))
+    case 'dimension':
+    case 'volume':
+    case 'weight':
+      return getMeasurment(value, locale)
+    case 'rating':
+      const { scale_max: length, value: val } = JSON.parse(value)
+      return Array.from({ length }, (_, i) =>
+        i <= val - 1 ? '&#9733;' : '&#9734;'
+      ).join('')
+    case 'color':
+      return `<figure style="background-color: ${value}; width: 1rem; height:1rem; display:block; margin-top: 2px; border-radius: 100%;"/>`
+    case 'url':
+      return `<a href="${value}" target="_blank" rel="norreferrer">${value}</a>`
+    case 'multi_line_text_field':
+      return value
+        .split('\n')
+        .map((line) => `${line}<br/>`)
+        .join('')
+    case 'json':
+    case 'single_line_text_field':
+    case 'product_reference':
+    case 'page_reference':
+    case 'variant_reference':
+    case 'file_reference':
+    default:
+      return value
+  }
+}
```

Comment:
If we're doing some custom logic over the metafields, would it be possible to only use one value instead of having both `value` and `valueHtml`?

Reply:
I left the value to be the original provider value if the user wants to do his parsing to be able to have the original value. Same we have `description` & `descriptionHtml`. At Shopify, that value most times is a JSON stringified.

Reply:
What do you think are good use cases for this? because I think it's better to have transformers as utils for cases like this.

Reply:
For example, the case was to have the auto-transformed value of the custom field. Let's take the weight as an example. The value that comes from Shopify is `{unit: MILLIMETERS, value: 10}`, and it wanted to provide the auto-transformed value `10mm` (or a link, rating, date, etc.), but if the user wants the original value for his transformation, to can access it, and not modify the provider for that. Yes agree with utils, but it is better to transform right after fetch and normalization, then call that util from the site folder, then conditionally call & include that since not all providers will have that

Reply:
Initially, I wanted to provide a component, parse, and conditionally show there. Since that would be used just by Shopify and create confusion, I decided on parsing on normalizer. Let me know which direction should we decide.